sklearn/feature_selection/sequential_feature_selector.py

```diff
@@ -0,0 +1,332 @@
+    --------
+    RFE : Recursive feature elimination selection of the best
+    number of features
+
+    References
+    ----------
+    .. [1] Ferri, F. J., Pudil P., Hatef, M., Kittler, J., "Comparative
+           study of techniques for large-scale feature selection,"
+           Pattern Recognition in Practice IV : 403-413, 1994
+
+    """
+    def __init__(self, estimator, n_features_to_select=1,
+                 forward=True, scoring=None,
+                 cv=5, n_jobs=1,
+                 pre_dispatch='2*n_jobs'):
+
+        self.estimator = estimator
+        self.n_features_to_select = n_features_to_select
+        self.forward = forward
+        self.pre_dispatch = pre_dispatch
+        self.scoring = scoring
+        self.cv = cv
+        self.n_jobs = n_jobs
+
+    def fit(self, X, y):
+        """Perform feature selection and learn model from training data.
+
+        Parameters
+        ----------
+        X : array-like, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+        y : array-like, shape = [n_samples]
+            Target values.
+
+        Returns
+        -------
+        self : object
+
+        """
+
+        if not (isinstance(self.n_features_to_select, int) or
+                isinstance(self.n_features_to_select, tuple)):
+            raise ValueError('n_features_to_select must be a positive integer'
+                             ' or tuple')
+
+        if isinstance(self.n_features_to_select, int) and (
+                    self.n_features_to_select < 1 or
+                    self.n_features_to_select > X.shape[1]):
+            raise ValueError('n_features_to_select must be a positive integer'
+                             ' between 1 and X.shape[1], got %s'
+                             % (self.n_features_to_select, ))
+
+        if isinstance(self.n_features_to_select, tuple):
+            if len(self.n_features_to_select) != 2:
+                raise ValueError('n_features_to_select tuple must consist of 2'
+                                 ' elements a min and a max value.')
+
+            if self.n_features_to_select[0] not in range(1, X.shape[1] + 1):
+                raise ValueError('n_features_to_select tuple min value must'
+                                 ' be in range(1, X.shape[1]+1).')
+
+            if self.n_features_to_select[1] not in range(1, X.shape[1] + 1):
+                raise ValueError('n_features_to_select tuple max value must'
+                                 ' be in range(1, X.shape[1]+1).')
+
+            if self.n_features_to_select[0] > self.n_features_to_select[1]:
+                raise ValueError('The min n_features_to_select value must be'
+                                 ' smaller than the max'
+                                 ' n_features_to_select value.')
+
+        if isinstance(self.n_features_to_select, tuple):
+            select_in_range = True
```

Comment:
Unless you really think it's necessary, I think having separate code paths for select_in_range just makes it harder to read. Can't we just set min and max to the int?

Reply:
yes, I will put that on my todo list. I would say, let's revisit that once all the unit tests are in place to make sure I don't mess anything up during the rewrite :P